ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java

```diff
@@ -97,12 +98,22 @@ public KStreamHolder<GenericKey> buildStream(
         consumedFactory
     );
 
-    final KStream<GenericKey, GenericRow> kstream = buildKStream(
-        source,
-        buildContext,
-        consumed,
-        nonWindowedKeyGenerator(source.getSourceSchema())
-    );
+    final KStream<GenericKey, GenericRow> kstream;
```

Comment:
I think this might be backwards incompatible - we should introduce a new `SourceBuilderV2` and use that going forward. Otherwise, if we don't add the `transformValues` for old queries the state store names will change as their position in the topology changes. I might be wrong (I don't remember whether we manually name our state stores or not) but we should double check on that.

Reply:
@agavra can you suggest some upgrade test options here?

Reply:
actually after looking it through I think we might be OK as this `addToTopology` flag is only called when it's a self-join; I may have just been confused. I think the historical plans would catch this if it were a bug.

Reply:
@agavra This only happens for self-joins so there is no backwards compatibility concern since self-joins are prohibited before this PR